src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+import duke.exception.InvalidCommandException;
+import duke.exception.InvalidIndexException;
+import duke.exception.InvalidInputException;
+import duke.message.Message;
+import duke.parser.UserInputParser;
+import duke.task.TaskList;
+
+/**
+ * Represents the Duke program.
+ */
+public class Duke {
+    private TaskList tasks;
+    private final Storage storage;
+    private final Ui ui;
+
+    /**
+     * Constructor for Duke.
+     */
+    public Duke() {
+        String filePath = "data/duke.txt";
+        this.ui = new Ui();
+        System.out.println(this.ui.getWelcomeMessage());
+        this.storage = new Storage(filePath);
+        // load data file
+        try {
+            this.tasks = new TaskList(storage.loadFile());
+        } catch (IOException e) {
+            System.out.println(this.ui.showError(e.getMessage()));
+            System.out.println(this.ui.getLine());
+            this.tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Main method.
+     * @param args Arguments.
```

Comment:
Would be good to have an empty line between description and parameter sections, according to the JavaDoc guide!